crates/consensus/src/transaction/mod.rs

```diff
@@ -47,20 +47,16 @@ pub trait Transaction: any::Any + Send + Sync + 'static {
     /// Get `gas_limit`.
     fn gas_limit(&self) -> u64;
 
-    /// Get `gas_price`.
-    fn gas_price(&self) -> Option<u128>;
-
     /// Returns the EIP-1559 the maximum fee per gas the caller is willing to pay.
     ///
     /// For legacy transactions this is `gas_price`.
     ///
     /// This is also commonly referred to as the "Gas Fee Cap".
     fn max_fee_per_gas(&self) -> u128;
 
-    /// Returns the EIP-1559 Priority fee the caller is paying to the block author.
-    ///
-    /// This will return `None` for non-EIP1559 transactions
-    fn max_priority_fee_per_gas(&self) -> Option<u128>;
+    /// Return the max priority fee per gas if the transaction is an EIP-1559 transaction, and
+    /// otherwise return the gas price.
+    fn max_priority_fee_per_gas(&self) -> u128;
```

Comment:
why is this no longer an option?

Reply:
removed redundancy within the consensus `Transaction` trait. `gas_price` has been deprecated in favour of `max fee per gas` and `max priority fee per gas`. checking if this returns an option is a bad way to check if this is a pre-eip1559 tx, and instead trait method `Transaction::ty` should be used.

Reply:
this should be an option because not every tx type has this